opal/mca/common/cuda/common_cuda.c

```diff
@@ -1797,25 +1798,25 @@ static int mca_common_cuda_is_gpu_buffer(const void *pUserBuf, opal_convertor_t
      * and set the current context to that.  It is rare that we will not
      * have a context. */
     res = cuFunc.cuCtxGetCurrent(&ctx);
-#endif /* OPAL_CUDA_GET_ATTRIBUTES */
     if (OPAL_UNLIKELY(NULL == ctx)) {
         if (CUDA_SUCCESS == res) {
-            res = cuFunc.cuPointerGetAttribute(&ctx,
+#if !OPAL_CUDA_GET_ATTRIBUTES
+            res = cuFunc.cuPointerGetAttribute(&memCtx,
```

Comment:
Does this overwrite the value of `memCtx`?  If so, what's the point of filling it in lines 1749 and 1754?  I ask because you specifically changed lines 1749 and 1754 to use the new variable `memCtx`...

More generally, after reading this patch a few times, I'm not sure what the difference is between `ctx` and `memCtx` -- it seems like they're used mutually exclusively, and therefore you don't really need 2 variables.  I don't really care, mind you -- this is your code, and if you want 2 variables just to make the code more readable, that's cool. :smile:  But I figured I'd ask: is there a reason to have 2 variables for the code logic?  Or is it just to make the code more readable?